--- ocs_mac_agent.php.orig	2008-06-12 06:58:34.511847656 +0200
+++ ocs_mac_agent.php	2008-06-12 06:56:02.990071830 +0200
@@ -1231,8 +1231,16 @@
                     $this->_profileData, $a_matches);
         // Can't find node...that's bad
         if (!$result && $in_required) {
-            $this->message("Could not find $in_nodeName in system_profiler XML.  Corrupted output?", 'error');
-            die();
+	   if ($in_nodeName == "SPSerialATADataType") {
+             // OS 10.3 does not return anything in this DataType if there is no motherboard support
+             // This should be nonfatal, so just return
+             $this->message("No " . $in_nodeName . " present.", 'info');
+             return '';
+	   }
+	   else {
+             $this->message("Could not find $in_nodeName in system_profiler XML.  Corrupted output?", 'error');
+             die();
+	   }
         }
         else if (!$result && !$in_required) {
             return '';
