--- /tmp/ocs_mac_agent.php	2010-06-30 13:32:36.000000000 +0200
+++ /usr/local/sbin/ocs_mac_agent.php	2010-06-30 13:31:06.000000000 +0200
@@ -784,7 +784,7 @@
         }
 
         $this->_OSXVer = preg_replace('/.*OS X (Server )?([\d\.]+)\s.*/', '\2', $this->_getValByKey('os_version', $this->_profileData));
-        $this->_is10_4 = ((strpos($this->_OSXVer, '10.4') === 0) || (strpos($this->_OSXVer, '10.5') === 0));
+        $this->_is10_4 = ((strpos($this->_OSXVer, '10.4') === 0) || (strpos($this->_OSXVer, '10.5') === 0) || (strpos($this->_OSXVer, '10.6') === 0) || (strpos($this->_OSXVer, '10.7') === 0));
         // Beginning of xml
         $this->_data['QUERY'] = 'INVENTORY';
         $this->_data['CONTENT']['DEVICEID'] = $this->_deviceId;
